building/glass/upload.py

```diff
@@ -123,6 +123,18 @@ def gs_rsync(local_path: str, remote_path: str, boto_path: str):
 """.lstrip()
 
 
+gs_branch_regex = re.compile(r'([a-z0-9_.-]+)((/[a-z0-9/_.-]*)?)')
+
+def gs_upload_path(branch):
+    m = gs_branch_regex.fullmatch(branch)
+    if m is None:
+        raise ValueError('invalid branch name {}'.format(branch))
+    domain, path = m.group(1), m.group(2)
+    if domain.endswith('.storage.googleapis.com'):
+        domain = domain[:-len('.storage.googleapis.com')]
```

Comment:
Any reason not to just use basic manipulation options?

i.e. `domain, path = branch.split('/', 1)` and then `if domain.endswith('.storage.googleapis.com'): domain = domain.rsplit('/', 3)[0]`

Reply:
the regex also validates the form of the URL